plugins/repository-s3/src/main/java/org/elasticsearch/plugin/repository/s3/S3RepositoryPlugin.java

```diff
@@ -85,4 +87,15 @@ public String description() {
     public void onModule(RepositoriesModule repositoriesModule) {
         repositoriesModule.registerRepository(S3Repository.TYPE, S3Repository.class, BlobStoreIndexShardRepository.class);
     }
+
+    public void onModule(SettingsModule module) {
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_AWS.KEY);
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_AWS.SECRET);
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_AWS.PROXY_PASSWORD);
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_S3.KEY);
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_S3.SECRET);
+        module.registerSettingsFilterIfMissing(AwsS3Service.CLOUD_S3.PROXY_PASSWORD);
+        module.registerSettingsFilter("access_key"); // WTF is this?
```

Comment:
I think it because you can define this per repository (so no prefix). Unsure though.
Got the same Q today while rewriting settings for the new infra.